homeassistant/components/huawei_lte/__init__.py

```diff
@@ -165,10 +165,20 @@ def get_data(key: str, func: Callable[[None], Any]) -> None:
                 )
                 self.subscriptions.pop(key)
             except ResponseErrorLoginRequiredException:
-                _LOGGER.info(
-                    "%s requires authorization, excluding from future updates", key
-                )
-                self.subscriptions.pop(key)
+                if isinstance(self.connection, AuthorizedConnection):
```

Comment:
Why is `get_data` defined inline ?

Reply:
Also, instead of all the if…else, can you replace that with `return` statements

Reply:
No particular reason for that inline I guess, can move even though it's not really related to this PR. Will also use a return to eliminate one `else`, even though that arguably makes the flow slightly muddier in this case with the `finally` block. But meh, matter of taste, I don't mind.